akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -253,11 +253,11 @@ object ByteString {
     }
 
     override def copyToArray[B >: Byte](dest: Array[B], start: Int, len: Int): Int = {
-      val copied = math.min(math.min(len, bytes.length), dest.length - start)
-      if (copied > 0) {
-        System.arraycopy(bytes, 0, dest, start, copied)
+      val toCopy = math.min(math.min(len, bytes.length), dest.length - start)
+      if (toCopy > 0) {
+        Array.copy(bytes, 0, dest, start, toCopy)
```

Comment:
This is the actual fix, same in ByteString1

Reply:
So, `arraycopy` and `Array.copy` have different semantics? That's subtle ;)

Reply:
I'm honestly quite surprised by this fixing the issue, what `Array.copy` contains is a check if the byte array is polymorpic/boxed and fall back to some slow copy in that case, but I'm not sure how that can happen with `ByteStrings` (at least not outside of the unit test, in the unit test it perhaps makes more sense that it could accidentally be a boxed byte array passed to a `ByteString` constructor or something like that)

Reply:
It seems more likely that the condition added above `&& pos < xs.length` would be the fix. Maybe this change here wasn't really necessary?

Reply:
No, the `pos < xs.length` is in the `ByteIterator` and that isn't used in 2.13, only 2.12 (and was a separate issue I discovered when adding test coverage, causing 2.12 getting stuck in an infinite loop). 2.13  has the `copyToArray`/`toArray` in ByteString never create an iterator at all.

Reply:
Ah, I see. Still changing that back makes no difference afaics.

Reply:
Same here, hmmmm. Ok, I see now, the actual fix is L400, where 0 was used instead of `startIndex`. 🤦‍♂

Reply:
I checked again. The actual fix (aside from ByteIterator) is using `startIndex` instead of `0` in line 400 and adding `start +` in line 657.

Reply:
Ah, yes, L657 as well. I'll do a PR later going back to System.arraycopy then...

Reply:
Maybe also rerun the benchmark just to be sure.

Reply:
Did some benchmarks yesterday but couldn't get any repeatable result that said System.arraycopy was faster than Array.copy so I'll leave it be (perhaps the check is jitted away because the array is never polymorphic).

Reply:
I asked not only because of that particular change but also because of the fixes. But I guess it shouldn't change anything because the same amount of data is copied just from the right offset.